File: backend/src/main/kotlin/com/study/restaurantapp/service/UserService.kt
package com.study.restaurantapp.service

import com.study.restaurantapp.model.LoginDto
import com.study.restaurantapp.model.TokenDto
import com.study.restaurantapp.model.User
import com.study.restaurantapp.model.UserDto

interface UserService {
    fun getAllUsers(): List<User>
    fun getUserById(id: String): User
    fun createUser(userDto: UserDto): User
    fun updateUser(id: String, userDto: UserDto): User
    fun deleteUser(id: String)
    fun refreshToken(tokenDto: TokenDto): String
    fun registerUser(userDto:UserDto)
    fun loginUser(loginDto: LoginDto): String
}

File: frontend/restaurant-app/src/app/user.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import {User} from "./models/user";


@Injectable({
  providedIn: 'root'
})
export class UserService {
  private apiUrl = 'http://localhost:8080/api';

  constructor(private http: HttpClient) { }

  getAllUsers(): Observable<User[]> {
    return this.http.get<User[]>(`${this.apiUrl}/users`);
  }

  getUserById(id: string): Observable<User> {
    return this.http.get<User>(`${this.apiUrl}/users/${id}`);
  }

  createUser(user: User): Observable<User> {
    return this.http.post<User>(`${this.apiUrl}/users`, user);
  }

  updateUser(id: string, user: User): Observable<User> {
    return this.http.put<User>(`${this.apiUrl}/users/${id}`, user);
  }

  deleteUser(id: string): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/users/${id}`);
  }

  loginUser(user: User): Observable<string> {
    return this.http.post<string>(`${this.apiUrl}/login`, user);
  }

  registerUser(user: User): Observable<string> {
    return this.http.post<string>(`${this.apiUrl}/register`, user);
  }
}
